components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts

```diff
@@ -243,14 +241,19 @@ export class PrebuildStatusMaintainer implements Disposable {
         const ctx = TraceContext.childContext("periodicUpdatableCheck", {});
 
         try {
-            const unresolvedUpdatables = await this.workspaceDB.trace(ctx).getUnresolvedUpdatables();
+            const unresolvedUpdatables = await this.workspaceDB.trace(ctx).getUnresolvedUpdatables(MAX_UPDATABLES);
             for (const updatable of unresolvedUpdatables) {
-                if (Date.now() - Date.parse(updatable.workspace.creationTime) > MAX_UPDATABLE_AGE) {
+                const age = Date.now() - Date.parse(updatable.workspace.creationTime);
+                if (age > MAX_UPDATABLE_AGE && age < IGNORED_UPDATEABLE_AGE) {
                     log.info(
                         "found unresolved updatable that's older than MAX_UPDATABLE_AGE and is inconclusive. Resolving.",
                         updatable,
                     );
-                    await this.doUpdate(ctx, updatable, updatable.prebuild);
+                    try {
+                        await this.doUpdate(ctx, updatable, updatable.prebuild);
+                    } catch (error) {
+                        log.error("Failed to process prebuild updatable.", error, { updatable });
```

Comment:
nit: is there opportunity to extract `logContext` info (owner, worspaceId, wsiId)from the updateable and pass it as first arg here?